dist/gmusic.js

```diff
@@ -239,6 +245,87 @@ proto.playback = {
   }
 };
 
+// Create a queue API
+proto.queue = {
+  _fireEvent: function (el, etype) {
+    if (el.fireEvent) {
+      el.fireEvent('on' + etype);
+    } else {
+      var evObj = document.createEvent('Events');
+      evObj.initEvent(etype, true, false);
+      el.dispatchEvent(evObj);
+    }
+  },
+
+  _render: function (container, force) {
+    // DEV: The queue isn't rendered till a click event is fired on this element
+    //      We must hide the queue during the 400ms animation and then reveal it
+    //      once both the 400ms in and 400ms out animations are complete
+    var table = container.querySelector('.queue-song-table');
+    if (container.style.display === 'none' && (!table || force)) {
+      // DEV: Hide the queue elements while we rapidly "render" the queue element
+      //      We have to use a style element because inline styles are saved by GPM
+      var style = document.createElement('style');
+      style.innerHTML = SELECTORS.queue.container + '{left: 10000px !important}';
+      document.body.appendChild(style);
+
+      // Render queue
+      this.queue._fireEvent(document.querySelector(SELECTORS.queue.triggerButton), 'click');
+      setTimeout(function () {
+        // Return queue to intitial state
+        this.queue._fireEvent(document.querySelector(SELECTORS.queue.triggerButton), 'click');
+        setTimeout(function () {
+          document.body.removeChild(style);
+        }, 20);
+      }.bind(this), 20);
+    }
+  },
+
+  clear: function (cb) {
+    var clearButton = this.doc.querySelector(SELECTORS.queue.container + ' ' + SELECTORS.queue.clearButton);
+    if (clearButton) {
+      clearButton.click();
+      setTimeout(function reRenderQueue () {
+        this.queue._render(this.doc.querySelector(SELECTORS.queue.container), true);
```

Comment:
The render call in `getSongs()` only renders if nothing has been rendered before (because GPM updates the queue even if it is hidden if the queue changes).

But for whatever reason it doesn't update when the clear button is clicked so we use the `force` parameter on the `_render` function to force GPM to update.